src/univalue/univalue_read.cpp

```diff
@@ -377,9 +377,11 @@ bool UniValue::read(const char *raw)
         default:
             return false;
         }
-    }
+    } while (!stack.empty ());
 
-    if (stack.size() != 0)
+    /* Check that nothing follows the initial construct (parsed above).  */
+    tok = getJsonToken(tokenVal, consumed, raw);
```

Comment:
Well, they are also used in the loop.  That's why I moved them above it.  Of course, we could also define them *both* locally in the loop *and* then here.  Would you prefer that?